docs/src/content/docs/cookbook/dexes/stonfi.mdx

```diff
@@ -5,10 +5,362 @@ sidebar:
+            value: GasSwapJettonToJetton,
+            body: JettonTransfer{
+                queryId: 42,
+                amount: offerAmount,
+                destination: RouterAddress,
+                responseDestination: myAddress(),
+                forwardTonAmount: GasSwapJettonToJettonFwd,
+                forwardPayload: forwardPayload.toCell(),
+            }.toCell()
+        });
+    }
+}
+```
+
+### Jetton to TON
+
+Swapping jetton to TON is similar to jetton/jetton swap. The only one difference is that `RouterJettonWallet` address should be replaced with `RouterProxyTonWallet`.
+
+```tact
+const RouterAddress: Address = address("kQALh-JBBIKK7gr0o4AVf9JZnEsFndqO0qTCyT-D-yBsWk0v"); // CPI Router v2.1.0
+const RouterProxyTonWallet: Address = address("kQBbJjnahBMGbMUJwhAXLn8BiigcGXMJhSC0l7DBhdYABhG7"); // Router's pTON Address
+
+message SwapJettonTon {
+    myJettonWalletAddress: Address;
+}
+
+contract JettonToTon {
+    receive() {}
+    receive(msg: SwapJettonTon) {
+        // This example directly uses the `myJettonWalletAddress` provided in the received message.
+        // In real-world scenarios, it's more reliable to calculate this address on-chain or save it during initialization to ensure accuracy and trustworthiness.
+        let myJettonWalletAddress: Address = msg.myJettonWalletAddress;
+
+        let offerAmount: Int = 100000; // the quantity of jettons being swapped
+
+        let forwardPayload = StonfiSwap{
+            otherTokenWallet: RouterProxyTonWallet,
+            refundAddress: myAddress(),
+            excessesAddress: myAddress(),
+            deadline: now() + 10000,
+            additionalData: SwapAdditionalData{
+                minOut: 1, // swap will fail if a receiver should receive less than minOut of tokens as a result
+                receiver: myAddress(),
+            }
+        };
+
+        send(SendParameters{
+            to: myJettonWalletAddress,
+            value: GasSwapJettonToTon,
+            body: JettonTransfer{
+                queryId: 42,
+                amount: offerAmount,
+                destination: RouterAddress,
+                responseDestination: myAddress(),
+                forwardTonAmount: GasSwapJettonToTonFwd,
+                forwardPayload: forwardPayload.toCell(),
+            }.toCell()
+        });
+    }
+}
+```
+
+### TON to Jetton
+
+TON to jetton swap introduces `ProxyTonTransfer` cause all interaction is done with pTON contract.
+
+Jetton `pTON` is a specialized asset within the STON.fi ecosystem on the TON blockchain. It functions as an intermediary within the STON.fi ecosystem to manage and facilitate efficient TON transfers.
+
+```tact
+message(0x01f3835d) ProxyTonTransfer {
+    queryId: Int as uint64 = 0;
+    tonAmount: Int as coins;
+    refundAddress: Address;
+    forwardPayload: Cell?;
+}
+
+const RouterProxyTonWallet: Address = address("kQBbJjnahBMGbMUJwhAXLn8BiigcGXMJhSC0l7DBhdYABhG7"); // Router's pTON wallet Address
```

Comment:
Inline comments like this are tough to read, since the user has to horizontally scroll for those a lot. Especially on mobile.

The suggestion here is to make comments precede what they're describing, like done for DeDust cookbook page